cli/packages/cmd/agent.go

```diff
@@ -311,9 +312,24 @@ func ParseAgentConfig(configFile []byte) (*Config, error) {
 	return config, nil
 }
 
-func secretTemplateFunction(accessToken string, existingEtag string, currentEtag *string) func(string, string, string) ([]models.SingleEnvironmentVariable, error) {
-	return func(projectID, envSlug, secretPath string) ([]models.SingleEnvironmentVariable, error) {
-		res, err := util.GetPlainTextSecretsV3(accessToken, projectID, envSlug, secretPath, false, false, "")
+type secretArguments struct {
+	IsRecursive bool `json:"recursive"`
+}
+
+func secretTemplateFunction(accessToken string, existingEtag string, currentEtag *string) func(string, string, string, ...string) ([]models.SingleEnvironmentVariable, error) {
+	// ...string is because golang doesn't have optional arguments.
+	// thus we make it slice and pick it only first element
+	return func(projectID, envSlug, secretPath string, args ...string) ([]models.SingleEnvironmentVariable, error) {
+		var parsedArguments secretArguments
+		// to make it optional
+		if len(args) > 0 {
+			err := json.Unmarshal([]byte(args[0]), &parsedArguments)
```

Comment:
but what if there are more arguments? How will that work? Maybe we should assign each arg number to a field so it is clear how to add other args in the future?

Reply:
What do you mean other args it will be simply a json object.

{recursive: true, arg2: value ...}

Reply:
I guess a better question is how will the unmarshal work when there are more than one arg? I see you doing arg[0]. Will we have a bunch of if statments to unmarshal each arg?